### ISSUE TITLE
[workspaces] conan install fails after upgrading from 1.9.2 to 1.10.0

### ISSUE THREAD

**Issue by petermbauer** [conan-io/conan#4271]
hi,

i have a workspaces setup here which works up to Conan 1.9.2 but fails with Conan >= 1.10.0.
I know this is an experimental feature but i could not find a hint in the release notes regarding breaking changes so i hope for hints/ideas.

thx

```
λ conan install ..
WARN: Migration: Updating settings.yml
WARN: ****************************************
WARN: settings.yml is locally modified, can't be updated
WARN: The new settings.yml has been stored in: C:\Users\XXXX\.conan\settings.yml.new
WARN: ****************************************
c:\users\XXXX\appdata\local\programs\python\python36\lib\site-packages\conans\util\fallbacks.py:9: UserWarning: Provide the output argument explicitly to function 'conans.client.tools.files.replace_in_file'
  warnings.warn("Provide the output argument explicitly{}".format(fn_str))
WARN: replace_in_file didn't find pattern '[plugins]' in 'C:\Users\XXXX\.conan\conan.conf' file.
Using conanws.yml file from C:\Users\XXXX\workspaces\test\conan-workspace
libxml2/2.9.8@us/develop: WARN: Warning! Static builds in Windows are unstable
Traceback (most recent call last):
  File "c:\users\XXXX\appdata\local\programs\python\python36\lib\site-packages\conans\model\ref.py", line 116, in loads
    _, name, version, user, channel, revision, _ = ConanFileReference.sep_pattern.split(text)
ValueError: not enough values to unpack (expected 7, got 1)

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "c:\users\XXXX\appdata\local\programs\python\python36\lib\site-packages\conans\client\command.py", line 371, in install
    reference = ConanFileReference.loads(args.path_or_reference)
  File "c:\users\XXXX\appdata\local\programs\python\python36\lib\site-packages\conans\model\ref.py", line 119, in loads
    "OpenCV/1.0.6@user/stable" % text)
conans.errors.ConanException: Wrong package recipe reference ..
Write something like OpenCV/1.0.6@user/stable

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "c:\users\XXXX\appdata\local\programs\python\python36\lib\site-packages\conans\client\command.py", line 1427, in run
    method(args[0][1:])
  File "c:\users\XXXX\appdata\local\programs\python\python36\lib\site-packages\conans\client\command.py", line 382, in install
    install_folder=args.install_folder)
  File "c:\users\XXXX\appdata\local\programs\python\python36\lib\site-packages\conans\client\conan_api.py", line 87, in wrapper
    return f(*args, **kwargs)
  File "c:\users\XXXX\appdata\local\programs\python\python36\lib\site-packages\conans\client\conan_api.py", line 518, in install
    manager.install_workspace(graph_info, workspace, remote_name, build, update)
  File "c:\users\XXXX\appdata\local\programs\python\python36\lib\site-packages\conans\client\manager.py", line 35, in install_workspace
    workspace)
  File "c:\users\XXXX\appdata\local\programs\python\python36\lib\site-packages\conans\client\graph\graph_manager.py", line 141, in load_graph
    processed_profile=processed_profile)
  File "c:\users\XXXX\appdata\local\programs\python\python36\lib\site-packages\conans\client\graph\graph_manager.py", line 224, in _load_graph
    graph = builder.load_graph(root_node, check_updates, update, remote_name, processed_profile)
  File "c:\users\XXXX\appdata\local\programs\python\python36\lib\site-packages\conans\client\graph\graph_builder.py", line 39, in load_graph
    processed_profile)
  File "c:\users\XXXX\appdata\local\programs\python\python36\lib\site-packages\conans\client\graph\graph_builder.py", line 105, in _load_deps
    remote_name, processed_profile)
  File "c:\users\XXXX\appdata\local\programs\python\python36\lib\site-packages\conans\client\graph\graph_builder.py", line 105, in _load_deps
    remote_name, processed_profile)
  File "c:\users\XXXX\appdata\local\programs\python\python36\lib\site-packages\conans\client\graph\graph_builder.py", line 131, in _load_deps
    if self._recurse(closure, new_reqs, new_options):
  File "c:\users\XXXX\appdata\local\programs\python\python36\lib\site-packages\conans\client\graph\graph_builder.py", line 154, in _recurse
    if n and self._conflicting_references(n.conan_ref, req.conan_reference):
  File "c:\users\XXXX\appdata\local\programs\python\python36\lib\site-packages\conans\client\graph\graph_builder.py", line 141, in _conflicting_references
    assert(previous_ref.revision is not None)
AssertionError

ERROR:

```

- [x] I've read the [CONTRIBUTING guide](https://github.com/conan-io/conan/blob/develop/.github/CONTRIBUTING.md).
- [x] I've specified the Conan version, operating system version and any tool that can be relevant.
- [x] I've explained the steps to reproduce the error or the motivation/use case of the question/suggestion.

**Comment by ytimenkov** [comment]
Could be related to stricter check of reference format.
Before it was possible to have reference with only slashes, i.e. "name/version/user/channel".
Now "@" is strictly required.

**Comment by petermbauer** [comment]
thx, this seems to be caused by the changes made with #4044 but the exception is not about a specific package but about `..`:
```
conans.errors.ConanException: Wrong package recipe reference ..
Write something like OpenCV/1.0.6@user/stable
```
So i don`t know what to do about it.
Is there any chance this will be fixed or shall i wait for the new implementation based on the "editable packages" feature?

**Comment by petermbauer** [comment]
As the "Editable Packages" feature has now been released with Conan 1.12: please confirm that this issue will not be fixed so i won't wait for it but migrate to the new approach.
